airbyte-integrations/connectors/source-tyntec-sms/source_tyntec_sms/tyntec_sms.yaml

```diff
@@ -0,0 +1,59 @@
+version: "0.1.0"
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: []
+  requester:
+    # API Docs: https://api.tyntec.com/reference/sms/current.html#sms-api
+    url_base: "https://api.tyntec.com"
+    http_method: "GET"
+    authenticator:
+      type: ApiKeyAuthenticator
+      header: "apikey"
+      api_token: "{{ config['api_key'] }}"
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
+    requester:
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  # API Docs: https://api.tyntec.com/reference/sms/current.html#sms-api-Read%20SMS%20status
+  messages_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "messages"
+      primary_key: "doneDate"
+      path: "/messaging/v1/messages/{{ config['request_id'] }}"
```

Comment:
Why you don't use the other endpoint to collect all `request_id` messgages?
https://api.tyntec.com/reference/sms/current.html#sms-api-Send%20SMS%20(GET)

Are you sure the primary key is `DoneDate`?

Reply:
@marcosmarxm Are you suggesting the linked endpoint should be used to create a new stream to get `request_id`, which would then be passed onto the `messages` stream as a parameter? If so, how could I configure such a stream and the `messages` stream so they link up properly?

As for the primary key being `doneDate`, the response doesn't have a designated key so I picked that as a reasonable choice. Let me know if you think a different field (or even no field) is more suitable from the response fields: https://api.tyntec.com/reference/sms/current.html#sms-api-restmessageresponse

Reply:
@marcosmarxm Are you suggesting the linked endpoint is to be used to create a new stream to get `request_id`, which would then be passed onto the `messages` stream as a parameter? If so, how could I configure such a stream and the `messages` stream so they link up properly? The linked endpoint also seems to be designed to send single SMS messages and return the `request_id` of the sent message, so I don't know if it can retrieve all `request_id` values of previously sent messages. I suppose if we configure it so that every SMS message is sent through this new stream and its `request_id` is stored or immediately used to trigger the `messages` stream, that is a viable alternative.

As for the primary key being `doneDate`, the response doesn't have a designated key so I picked it as a reasonable choice. Let me know if you think a different field (or even no field) is more suitable from the response fields: https://api.tyntec.com/reference/sms/current.html#sms-api-restmessageresponse

Reply:
@marcosmarxm Are you suggesting the linked endpoint is to be used to create a new stream to get `request_id`, which would then be passed onto the `messages` stream as a parameter? If so, how could I set up such a stream and the `messages` stream so they link up properly? The linked endpoint also seems to be designed to send a single SMS message and return the `request_id` of the sent message, so I don't know if it can fetch `request_id` values of previously sent messages. I suppose we can make it work if we configure it so every SMS message is sent via this new stream and the `request_id` is stored or immediately used to trigger the `messages` stream.

As for the primary key being `doneDate`, the response doesn't have a designated key so I picked it as a reasonable choice. Let me know if you think a different field (or even no field) is more suitable from the response fields: https://api.tyntec.com/reference/sms/current.html#sms-api-restmessageresponse

Reply:
Exactly. Please read the docs about StreamSlicer: https://docs.airbyte.com/connector-development/config-based/understanding-the-yaml-file/stream-slicers check other connectors implementation using ListStreamSlicer.

Reply:
Check Survey Sparrow connector, it uses a List StreamSlicer can help you.